crates/solver/src/driver.rs

```diff
@@ -145,13 +148,70 @@ impl Driver {
         .await
     }
 
+    async fn optimize_cost_of_unwrapping(&self, settlement: &mut Settlement) -> Result<()> {
```

Comment:
I would like to avoid adding this logic to `driver.rs` as it is already little bit our "messy cellar" where we have a lot of different logic together.
I was originally thinking that this kind of "settlement post processing" could be it's own component/pipeline where we can add multiple optimisations in a row. With this being our first one, it might be a bit too early to build an abstraction (we don't really know yet what other processing steps we will have, if any), but maybe we can move this logic into it's own submodule (settlement_post_processing or the like).

Also, I feel this method doesn't necessarily require a e2e test, but could well be covered by unit tests (given that the settlement is supposed to be pretty much a data struct and we are "just" doing some logically involved modifications to it). Currently the difficult part for a unit test is that we need to know the buffer which requires an async call on an unmockable web3 instance.

We already have a `BufferRetrieving` trait which is used by the `http_solver` to pass internal buffers into our MiP and Quasimodo solver. This trait could be extended to also support the native token (right now it's limited to ERC20) and then used here. We could then easily mock it and have unit tests covering all of the scenarios.

This is also brings the small performance advantage that we could create a cache of buffers "per settlement run" to avoid double fetching these balances (not needed for this PR).